cockatrice/src/pixmapgenerator.cpp

```diff
@@ -112,26 +112,35 @@ QPixmap CountryPixmapGenerator::generatePixmap(int height, const QString &countr
 
 QMap<QString, QPixmap> CountryPixmapGenerator::pmCache;
 
-QPixmap UserLevelPixmapGenerator::generatePixmap(int height, UserLevelFlags userLevel, bool isBuddy)
+QPixmap UserLevelPixmapGenerator::generatePixmap(int height, UserLevelFlags userLevel, bool isBuddy, QString privLevel)
 {
-
-    int key = height * 10000 + (int) userLevel + (int) isBuddy;
+    int privkeysize = privLevel.size();
+    if (privLevel.size() == 0)
+        privkeysize = 1;
+    
+    int key = height * 10000 + (int)userLevel * 88 + (int)isBuddy * 54 + privkeysize * 31;
```

Comment:
These magic numbers are quite hard to understand; also, using the length of `privLevel` as a key will cause collisions if in the future we'll add new levels.
Maybe a better idea is to change the `pmCache` map to use string keys:
```cpp
QMap<QString, QPixmap> UserLevelPixmapGenerator::pmCache;
```
and then generate the key like this:
```cpp
QString key = QString::number(height * 10000 + (int) userLevel + (int) isBuddy) + privLevel;
```